### PR TITLE
Delete /tmp/setvendordb file when done

### PR DESCRIPTION
**Pull request by hallyn**
Hopefully we are in fact done with it after this function...  Right now every time I create a keyset one of these files leaks.

### CODE REVIEW

**Review by smoser — APPROVED** [review]
gracias.

File: go/pkg/shim/shim.go [hunk 1]
```diff
@@ -73,6 +73,7 @@ func SetVendorDB(shim string, db, dbx efi.SignatureDatabase) error {
 	if err != nil {
 		return err
 	}
+	defer os.Remove(fp.Name())
```
Comment (smoser) [inline go/pkg/shim/shim.go:77]: c-i says you need to import os

> pkg/shim/shim.go:76:8: undefined: os

Comment (hallyn) [reply go/pkg/shim/shim.go:77]: Thank you, CI.

**Comment by hallyn** [comment]
No, this can't be right.  We set fp.Name() in the section, not the contents of fp.

So we can't delete it, the caller has to.

**Comment by hallyn** [comment]
Oh, obj.SetSections() does the objcopy, so it's actually fine.

BTW, probably the more go-ish thing to do in 'obj' pkg is to make an array of Sections a type of its own, make SetSections() a function on that type, and then just call 

```
sections.Set()
```

instead of

```
obj.SetSections(shim, sections...)
```

**Comment by smoser** [comment]
> Oh, obj.SetSections() does the objcopy, so it's actually fine.
> 
> BTW, probably the more go-ish thing to do in 'obj' pkg is to make an array of Sections a type of its own, make SetSections() a function on that type, and then just call
> 
> ```
> sections.Set()
> ```
> 
> instead of
> 
> ```
> obj.SetSections(shim, sections...)
> ```

i'm not opposed i guess.

File: subs.mk [hunk 2]
```diff
@@ -2,7 +2,7 @@ KEYSET ?= snakeoil
 DOCKER_BASE ?= docker://
 UBUNTU_MIRROR ?= http://archive.ubuntu.com/ubuntu
 KEYSET_D ?= $(HOME)/.local/share/machine/trust/keys/$(KEYSET)
-MOSCTL_BINARY ?= https://github.com/project-machine/mos/releases/download/v0.0.14/mosctl
+MOSCTL_BINARY ?= /home/serge/src/project-machine/mos/mosctl
```
Comment (smoser) [inline subs.mk]: whoops!

File: go/pkg/shim/shim.go [hunk 3]
```diff
@@ -73,6 +74,7 @@ func SetVendorDB(shim string, db, dbx efi.SignatureDatabase) error {
...
 
 	if err := VendorDBSectionWrite(fp, db, dbx); err != nil {
 		return err
```
Comment (smoser) [inline go/pkg/shim/shim.go:81]: While you're here, could you fix the filehandle leak in the error path here?

ie, fp.Close() never gets called if we take this error path. (it gets called in the happy path one line below).

File: subs.mk (hunk 2 above)
Comment (hallyn) [reply subs.mk]: Oh no - I told myself not to add that file to the index but i guess i ended up git add -a'ing after all.

**Review by smoser — APPROVED** [review]

**Pull request merged**